src/pke/lib/scheme/ckksrns/ckksrns-parametergeneration.cpp

```diff
@@ -78,7 +78,9 @@ bool ParameterGenerationCKKSRNS::ParamsGenCKKSRNS(std::shared_ptr<CryptoParamete
     uint32_t qBound        = firstModSize + (numPrimes - 1) * scalingModSize + extraModSize;
 
     // we add an extra bit to account for the alternating logic of selecting the RNS moduli in CKKS
-    qBound++;
+    // ignore the case when there is only one max size modulus
+    if (qBound != static_cast<double>(auxBits))
```

Comment:
qBound and auxBits are of the same type - uint32_t. The cast to double is not needed.